test/e2e/framework/addon/vault/vault.go

```diff
@@ -230,6 +225,22 @@ func (v *Vault) Setup(cfg *config.Config, leaderData ...internal.AddonTransferab
 		)
 	}
 
+	// Set E2E_OPENSHIFT=true if you're running the E2E tests against an OpenShift
+	// cluster.
+	// OpenShift requires some different settings. See
+	// https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-openshift
+	if os.Getenv("E2E_OPENSHIFT") == "true" {
+		v.chart.Vars = append(
+			v.chart.Vars,
+			[]chart.StringTuple{
+				{
+					Key:   "global.openshift",
+					Value: "true",
+				},
+			}...,
```

Comment:
Without this setting the Vault chart fails to install:

```
Release "chart-vault-vault" does not exist. Installing it now.
W1005 13:06:54.265120    6895 warnings.go:70] would violate PodSecurity "restricted:v1.24": unrestricted capabilities (container "vault" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "vault" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
NAME: chart-vault-vault
```

Reply:
@wallrj could we instead set the capabilities.drop = "ALL" and seccompProfile RuntimeDefault regardless of whether we are running on openshift?

Reply:
We could,  but the helm chart makes other changes to the manifests  contingent on the `global.openshift` value.

https://github.com/hashicorp/vault-helm/blob/e2990d291319b2097c9366918078263343173755/templates/_helpers.tpl#L545-L579

The advantage of my way is that we're using the chart as the authors intended, which makes the test more realistic and ensures that we will get any future OpenShift improvements that are made in the chart  and we will be using the chart recommended defaults on our E2E Kind cluster.

The advantage of your suggestion is that we wouldn't need the `E2E_OPENSHIFT` flag.

I will leave it like this for now.